src/dmd/cond.d

```diff
@@ -362,13 +375,56 @@ extern (C++) final class StaticForeach : RootObject
         auto catass = new CatAssignExp(aloc, new IdentifierExp(aloc, idres), res[1]);
         s2.push(createForeach(aloc, pparams[1], new ExpStatement(aloc, catass)));
         s2.push(new ReturnStatement(aloc, new IdentifierExp(aloc, idres)));
-        auto aggr = wrapAndCall(aloc, new CompoundStatement(aloc, s2));
-        sc = sc.startCTFE();
-        aggr = aggr.expressionSemantic(sc);
-        aggr = resolveProperties(sc, aggr);
-        sc = sc.endCTFE();
-        aggr = aggr.optimize(WANTvalue);
-        aggr = aggr.ctfeInterpret();
+
+        Expression aggr;
+        Type indexty;
+
+        if (rangefe && (indexty = ety.typeSemantic(aloc, sc)).isintegral())
+        {
+            rangefe.lwr.type = indexty;
+            rangefe.upr.type = indexty;
+            auto lwrRange = getIntRange(rangefe.lwr);
+            auto uprRange = getIntRange(rangefe.upr);
+
+            const lwr = rangefe.lwr.toInteger();
+            auto  upr = rangefe.upr.toInteger();
+            size_t length = 0;
+
+            if (lwrRange.imin <= uprRange.imax)
+            {
+                auto bounds = IntRange(SignExtendedNumber(size_t.min), SignExtendedNumber(size_t.max));
```

Comment:
Just checked back again.  I really can't stress enough that error messages should not be issued due to limitations of the host.  Either the target is unable to execute, or we don't issue an error at all.

Reply:
In that case I will just remove the error branch.
length will wrap but that feels more in line with the compiler.